lib/chipmunk_bagger.rb

```diff
@@ -0,0 +1,67 @@
+require 'chipmunk_bag'
+require 'chipmunk_metadata_error'
+require 'find'
+require 'bagit'
+require 'securerandom'
+require 'nokogiri'
+
+class ChipmunkBagger
+  
+  attr_accessor :content_type, :external_id, :src_path, :bag_path
+
+  def initialize(content_type, external_id, src_path, bag_path)
+    @content_type = content_type
+    @external_id = external_id
+    # make sure src_path ends in a '/'
+    src_path += '/' unless src_path[-1] == '/'
+    @src_path = src_path
+    @bag_path = bag_path
+  end
+
+
+  # Moves data from src_path to bag_path/data and generates appropriate manifests
+  def make_bag
+    raise ArgumentError, "Making bags for #{content_type} is not supported" unless content_type == 'audio'
+
+    # make a new bag with the given external id and content type at given path
+    @bag = ChipmunkBag.new bag_path
+    
+    # move everything into the data subdir if data subdir does not exist
+    move_files_to_data
+
+    bag.write_chipmunk_info(common_tags.merge(audio_tags))
+    bag.download_metadata
+
+    # generate the manifest and tagmanifest files
+    bag.manifest!
+  end
+
+  private
+
+  attr_accessor :bag
+
+  def common_tags
+    {
+      'External-Identifier' => external_id,
+      'Chipmunk-Content-Type' => content_type,
+      'Bag-ID' => SecureRandom.uuid
+    }
+  end
+
+  def move_files_to_data
+    Find.find(src_path) do |file|
+      bag.add_file_by_moving(file.sub(src_path,''),file) if File.file?(file)
```

Comment:
I'd like to see a temporary variable or method here for the `data`-relative file path. Maybe only because it was confusing to me to parse this out, but these two lines are very dense with a lot of differing uses of the both `file` and `src_path`.

Reply:
OK, I added comments and a helper method to make this clearer.